%!TEX root = ../Testprotokolle.tex
\chapter{Testdurchführung (2015-07-12)}

Art des Tests: Integrationstest\\
Testfall: T1100\\
Beteiligte Tester: Fabio Mazzone \& Gabriel Ahlers\\

Folgende Methode wird zum initializieren der Testumgebung vor jedem der Tests durchgeführt:\\\\
\hspace*{0mm}@BeforeClass \\
\hspace*{0mm}public static void app()\{ \\
\hspace*{3mm}FakeApplication fakeApplication = fakeApplication(inMemoryDatabase(\grqq test\grqq)); \\
\hspace*{3mm}start(fakeApplication); \\
\hspace*{3mm}Potion.init(); \\
\hspace*{0mm}\} \\\\

\section{Testprotokoll}
\begin{longtable}{|p{4cm}|p{11cm}|}
\hline
\textbf{Tester} & \textit{Gabriel Ahlers \& Fabio Mazzone} \\
\hline
\textbf{Eingaben} & \hspace*{0mm}@Test \newline
\hspace*{0mm}public void testInventoryCreate()\{ \newline
\hspace*{3mm}Profile profile = null; \newline
\hspace*{3mm}try \{ \newline
\hspace*{6mm}Profile profile = User.create("UserCreate", \grqq UserCreate@local.de\grqq, \grqq empty\grqq).profile; \newline
\hspace*{3mm}\} catch (UsernameTakenException | EmailTakenException e) \{ \newline
\hspace*{6mm}e.printStackTrace(); \newline
\hspace*{3mm}\} \newline
\hspace*{3mm}Potion potion = Potion.getById(1L); \newline
\hspace*{3mm}Inventory inv = Inventory.create(profile, potion); \newline
\hspace*{3mm}assertTrue(inv != null); \newline
\hspace*{3mm}Logger.info(\grqq InventoryCreate successfull\grqq); \newline
\hspace*{0mm}\} \\
\hline
\textbf{Soll - Reaktion} & \textit{Konsole zeigt Meldung: \grqq InventoryCreate successfull\grqq} \\
\hline
\textbf{Ist -- Reaktion} & \textit{Konsole zeigt Meldung: \grqq InventoryCreate successfull\grqq} \\
\hline
\textbf{Ergebnis} & \textit{Test erfolgreich durchgeführt} \\
\hline
\textbf{Unvorhergesehene Ereignisse} &
\textit{--} \\
\hline
\textbf{Nacharbeiten} & \textit{--} \\
\hline
\end{longtable}

\section{Zusammenfassung}

Der Testlauf ergab keine Fehler.